exec/java-exec/src/main/java/org/apache/drill/exec/store/dfs/WorkspaceSchemaFactory.java

```diff
@@ -151,15 +152,11 @@ public WorkspaceSchemaFactory(
   public boolean accessible(final String userName) throws IOException {
     final FileSystem fs = ImpersonationUtil.createFileSystem(userName, fsConf);
     try {
-      // We have to rely on the listStatus as a FileSystem can have complicated controls such as regular unix style
-      // permissions, Access Control Lists (ACLs) or Access Control Expressions (ACE). Hadoop 2.7 version of FileSystem
-      // has a limited private API (FileSystem.access) to check the permissions directly
-      // (see https://issues.apache.org/jira/browse/HDFS-6570). Drill currently relies on Hadoop 2.5.0 version of
-      // FileClient. TODO: Update this when DRILL-3749 is fixed.
-      fs.listStatus(wsPath);
+      fs.access(wsPath, FsAction.READ);
     } catch (final UnsupportedOperationException e) {
-      logger.trace("The filesystem for this workspace does not support this operation.", e);
+      logger.debug("The filesystem for this workspace does not support this operation.", e);
```

Comment:
I am not returning false to be consistent with the existing behavior.  If the user does not have permission, we fail when we try to read the file during execution.